forum/qa-plugin/tips-widget/tips-widget-widget.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+class tips_widget_widget
+{
+	public $random;
+	private $tips_list;
+	
+	function __construct()
+	{
+		if (!qa_opt('tips-enable'))
+			return;
+		
+		$this->tips_list = explode('!NEW!', qa_opt('tips-widget-content'));
+		$prev_random = isset($_COOKIE['prev_random']) ? $_COOKIE['prev_random'] : -1;
```

Comment:
sprawdzasz czy jest 'prev_random' ale nie sprawdzasz czy ma zawartość. poza tym nie walidujesz jej jakoś specjalnie, co moze byc niebezpieczne dla serwera

Reply:
Racja, poprawię to.